tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/AssembleRepositoryMojo.java

```diff
@@ -349,8 +353,9 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                             .flatMap(tpFile -> tpFile.getLocations().stream())
                             .filter(InstallableUnitLocation.class::isInstance).map(InstallableUnitLocation.class::cast)
                             .flatMap(iu -> iu.getRepositories().stream())
-                            .filter(iuRepo -> autoReferencesFilter.test(iuRepo.getLocation()))
-                            .map(iuRepo -> new RepositoryReference(null, iuRepo.getLocation(), true))
+                            .filter(iuRepo -> autoReferencesFilter.test(varResolver.resolve(iuRepo.getLocation())))
+                            .map(iuRepo -> new RepositoryReference(null, varResolver.resolve(iuRepo.getLocation()),
+                                    true))
```

Comment:
Lets save the second variable substitution.
```suggestion
                            .map(iuRepo -> varResolver.resolve(iuRepo.getLocation())).filter(autoReferencesFilter)
                            .map(location -> new RepositoryReference(null, location, true))
```